vizro-core/tests/unit/vizro/models/_components/test_button.py

```diff
@@ -53,9 +37,13 @@ def test_set_action_via_validator(self):
 
 
 class TestBuildMethod:
-    def test_button_build(self, expected_button):
-        button = vm.Button(id="button_id", text="Click me!").build()
-        result = json.loads(json.dumps(button, cls=plotly.utils.PlotlyJSONEncoder))
-        expected = json.loads(json.dumps(expected_button, cls=plotly.utils.PlotlyJSONEncoder))
-
-        assert result == expected
+    def test_button_build(self):
+        button = vm.Button(id="button_id", text="My text").build()
+        assert_component_equal(
+            button,
+            html.Div(
+                dbc.Button(id="button_id", children="My text", className="button_primary"),
+                className="button_container",
+                id="button_id_outer",
+            ),
```

Comment:
Any preference whether to input the object directly or store it in a variable called `expected` before?

Reply:
Hm, I think it would look easier to read if we actually stored it in `expected`.

Reply:
I would prefer not to be too prescriptive about this in general. I don't think we should insist on always having a variable called `expected` if the assert is very simple like `assert_component_equal(button, html.Div())`. Once the `expected` gets beyond a certain complexity, I think it's more readable to extract into a variable instead of having it inline, but the boundary of what that "certain complexity" is just a matter of taste really.

In this case I think I agree with @maxschulz-COL so I will extract it into a variable, but I don't think we need to require very simple cases to do that.

Worth noting that PyCharm handles the actual and expected in exactly the same way regardless of whether you extract it into a variable or not (left is expected, right is actual):
![image](https://github.com/mckinsey/vizro/assets/49395058/f4205bae-9736-4fbb-92a2-012196bf855d)